megamek/src/megamek/common/Compute.java

```diff
@@ -2520,41 +2520,36 @@ public static ToHitData getAttackerMovementModifier(Game game, int entityId,
             return toHit;
         }
 
+        int dedicatedGunnerMod = ((entity instanceof Mech mek) && (mek.getCockpitType() == Mech.COCKPIT_DUAL)
+                && entity.getCrew().hasDedicatedGunner()) ? 2 : 1;
+
         if ((entity.getMovementMode() == EntityMovementMode.BIPED_SWIM)
             || (entity.getMovementMode() == EntityMovementMode.QUAD_SWIM)) {
-            toHit.addModifier(3, "attacker used UMUs");
+            toHit.addModifier(3 / dedicatedGunnerMod, "attacker used UMUs");
         } else if (entity instanceof LandAirMech && movement == EntityMovementType.MOVE_VTOL_WALK) {
-            toHit.addModifier(3, "attacker cruised");
+            toHit.addModifier(3 / dedicatedGunnerMod, "attacker cruised");
         } else if (entity instanceof LandAirMech && movement == EntityMovementType.MOVE_VTOL_RUN) {
-            toHit.addModifier(4, "attacker flanked");
+            toHit.addModifier(4 / dedicatedGunnerMod, "attacker flanked");
         } else if ((movement == EntityMovementType.MOVE_WALK) || (movement == EntityMovementType.MOVE_VTOL_WALK)
                 || (movement == EntityMovementType.MOVE_CAREFUL_STAND)) {
-            toHit.addModifier(1, "attacker walked");
+            toHit.addModifier(1 / dedicatedGunnerMod, "attacker walked");
         } else if ((movement == EntityMovementType.MOVE_RUN) || (movement == EntityMovementType.MOVE_VTOL_RUN)) {
-            toHit.addModifier(2, "attacker ran");
+            toHit.addModifier(2 / dedicatedGunnerMod, "attacker ran");
         } else if (movement == EntityMovementType.MOVE_SKID) {
-            toHit.addModifier(3, "attacker ran and skidded");
+            toHit.addModifier(3 / dedicatedGunnerMod, "attacker ran and skidded");
         } else if (movement == EntityMovementType.MOVE_JUMP) {
             if (entity.hasAbility(OptionsConstants.PILOT_JUMPING_JACK)) {
-                toHit.addModifier(1, "attacker jumped");
+                toHit.addModifier(1 / dedicatedGunnerMod, "attacker jumped");
             } else if (entity.hasAbility(OptionsConstants.PILOT_HOPPING_JACK)) {
-                toHit.addModifier(2, "attacker jumped");
+                toHit.addModifier(2 / dedicatedGunnerMod, "attacker jumped");
             } else {
-                toHit.addModifier(3, "attacker jumped");
+                toHit.addModifier(3 / dedicatedGunnerMod, "attacker jumped");
```

Comment:
It looks like we can get non-integer results here, which I believe Java will round down.
1. is that expected behavior?
2. do we check anywhere that this code produces the expected results for every combination?

Reply:
1. I think so as it was like that before
2. No. Currently there are no combinations, this method only returns a single modifier. I did not check the rules as - see 1. :)